### ISSUE TITLE
Hard crash (process exit), something causes importing client entry during SSR

### ISSUE THREAD

**Issue by IlyaSemenov** [frandiox/vite-ssr#47]
This is a spin-off of https://github.com/frandiox/vite-ssr/issues/43 where I incorrectly identified the problem at first.

## Problem

In a small reproducible case, something causes importing client entry during SSR, which leads to hard crash (process exit). I had this on a larger project, but then simplified the setup to bare minimum for the reproduction.

## Reproduction

https://github.com/IlyaSemenov/vite-ssr-crash

```
~/tmp/4
❯ git clone https://github.com/IlyaSemenov/vite-ssr-crash.git
Cloning into 'vite-ssr-crash'...
remote: Enumerating objects: 26, done.
remote: Counting objects: 100% (26/26), done.
remote: Compressing objects: 100% (17/17), done.
remote: Total 26 (delta 5), reused 25 (delta 4), pack-reused 0
Receiving objects: 100% (26/26), 13.52 KiB | 157.00 KiB/s, done.
Resolving deltas: 100% (5/5), done.

~/tmp/4
❯ cd vite-ssr-crash

~/tmp/4/vite-ssr-crash master
❯ nodenv local 16.0.0

~/tmp/4/vite-ssr-crash master
❯ yarn
yarn install v1.22.10
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "vite-ssr > @rollup/plugin-replace@2.4.2" has unmet peer dependency "rollup@^1.20.0 || ^2.0.0".
warning "vite-ssr > react-ssr-prepass@1.4.0" has unmet peer dependency "react@^16.8.0 || ^17.0.0".
warning "vite-ssr > @rollup/plugin-replace > @rollup/pluginutils@3.1.0" has unmet peer dependency "rollup@^1.20.0||^2.0.0".
warning Workspaces can only be enabled in private projects.
[4/4] 🔨  Building fresh packages...
✨  Done in 1.69s.

~/tmp/4/vite-ssr-crash master
❯ yarn dev
yarn run v1.22.10
$ vite-ssr
Pre-bundling dependencies:
  vite-ssr
  vue
  vite-ssr/vue/entry-client
  vite-ssr/vue/entry-server
(this will be run only when your dependencies or config have changed)

  vite v2.3.8 dev server running at:

  > Local: http://localhost:3000/
  > Network: use `--host` to expose

 -- SSR mode

>>>>>>>>>>>>>> (in a separate console) curl localhost:3000

[Function (anonymous)]
TypeError: render is not a function
    at handleSsrRequest (/Users/semenov/tmp/4/vite-ssr-crash/node_modules/vite-ssr/dev/server.js:88:37)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
12:57:31 AM [vite] Internal server error: render is not a function
      at handleSsrRequest (/Users/semenov/tmp/4/vite-ssr-crash/node_modules/vite-ssr/dev/server.js:88:37)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
/node_modules/vite-ssr/vue/entry-client.js:20
    const url = window.location;
                ^

ReferenceError: window is not defined
    at Module.viteSSR [as default] (/node_modules/vite-ssr/vue/entry-client.js:20:17)
    at eval (/Users/semenov/tmp/4/vite-ssr-crash/src/main.ts:12:61)
    at instantiateModule (/Users/semenov/tmp/4/vite-ssr-crash/node_modules/vite/dist/node/chunks/dep-0ed4fbc0.js:69982:166)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

The two key points are:
* non-empty `alias` section in vite config
* importing (not even using - just importing and accessing) `onServerPrefetch`

____

**UPDATE:**

I came up with even simpler version without `onServerPrefetch`, which crashes the same way.

https://github.com/IlyaSemenov/vite-ssr-crash/tree/no_routes

```
git clone https://github.com/IlyaSemenov/vite-ssr-crash.git -b no_routes
```

**Comment by frandiox** [comment]
Thanks for the repro, I'll try to check it soon 👍

**Comment by frandiox** [comment]
So it was grabbing the first alias that contained `vite-ssr` for some internal logic, and it was finding your custom alias instead of the correct one. Should be fixed in `0.10.6` 👍
